pds_doi_core/util/doi_validator.py

```diff
@@ -0,0 +1,323 @@
+                logger.debug(f"VID_FIELD: {ii,rows[ii][col_index_to_vid]}")
+            lidvid = rows[last_row_index][col_index_to_lid] + '::' + rows[last_row_index][col_index_to_vid]
+            logger.debug(f"rows[last_row_index][col_index_to_doi] {rows[last_row_index][col_index_to_doi]}")
+            if doi.related_identifier == lidvid:   # If 'lidvid' field exist already with the same title, throw an exception.
+                col_index_to_title   = columns.index('title')
+
+                logger.debug(f"CHECK_FIELD_TITLE_EXIST_TRUE_LIDVID_EXIST_TRUE: doi: {rows[last_row_index][col_index_to_doi]}, title: '{doi.title}' lidvid: {lidvid} action: {action} force_flag: {force_flag}")
+                if action != 'release':
+                    if not force_flag:
+                        logger.error(f"The title: '{doi.title}' has already been used for a DOI by lidvid:{lidvid}, status: {rows[last_row_index][col_index_to_status]} , doi: {rows[last_row_index][col_index_to_doi]} . You must use a different title.")
+                        raise DuplicatedTitleDOIException(f"The title: '{doi.title}' has already been used for a DOI by lidvid:{lidvid}, status: {rows[last_row_index][col_index_to_status]} , doi: {rows[last_row_index][col_index_to_doi]}. You must use a different title.")
+                    else:
+                        if action == 'draft':
+                            logger.warning(f"The title: '{doi.title}' has already been used for a DOI by lidvid:{lidvid}, status: {rows[last_row_index][col_index_to_status]} , doi: {rows[last_row_index][col_index_to_doi]} but action is draft and force_flag is True.")
+                else:
+                    logger.debug(f"CHECK_FIELD_TITLE_EXIST_TRUE_LIDVID_EXIST_TRUE_ACTION_IS_RELEASE: doi: {rows[last_row_index][col_index_to_doi]}, title: '{doi.title}' ")
+            else:
+                # If the same title exist but no DOI minted yet, nothing to be done.
+                logger.debug(f"The title: '{doi.title}' exist but no DOI exist with lidvid: {doi.related_identifier} in local database.  No DOI has been minted.")
+                logger.debug(f"CHECK_FIELD_TITLE_EXIST_TRUE_LIDVID_EXIST_FALSE: title:'{doi.title}'")
+        else:
+            # If no title exist, nothing to be done.
+            logger.debug(f"The title: '{doi.title}' does not yet exist in local database.")
+            logger.debug(f"CHECK_FIELD_TITLE_EXIST_FALSE: title:'{doi.title}'")
+
+        return 1
+
+    def _check_field_title_content(self, doi: Doi,action):
+        """ Check if the pds4 label is a bundle the title should contain bundle (ignoring case)
+            The same for: dataset, collection, document
+            Otherwise we raise a warning.
+        """
+
+        if doi.product_type.lower() == 'bundle' and 'bundle' not in doi.title.lower():
+            logger.warning(f"A 'bundle' product type should contain 'Bundle' in the title.")
```

Comment:
We should raise an exception. You can name it TitleDoesNotMatchProductTypeException. The warning message are good.

Reply:
The new exception TitleDoesNotMatchProductTypeException has been added to pds_doi_core/input/exeptions.py file.